cdap-common/src/main/java/io/cdap/cdap/common/guice/RemoteAuthenticatorModules.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.common.guice;
+
+import com.google.inject.Module;
+import com.google.inject.PrivateModule;
+import com.google.inject.Scopes;
+import com.google.inject.name.Names;
+import com.google.inject.util.Providers;
+import io.cdap.cdap.common.internal.remote.DefaultRemoteAuthenticatorProvider;
+import io.cdap.cdap.common.internal.remote.RemoteAuthenticatorExtensionLoader;
+import io.cdap.cdap.security.spi.authenticator.RemoteAuthenticator;
+
+import javax.annotation.Nullable;
+
+/**
+ * Provides Guice bindings for {@link RemoteAuthenticator}.
+ */
+public class RemoteAuthenticatorModules {
```

Comment:
`final` util class. Also add a private constructor to prevent instantiation.

Reply:
Done